tests/data/autopygen/autopygen_end_to_end_sub_inputs.xml

```diff
@@ -0,0 +1,32 @@
+<section name="default" title="default" expanded="true">
+<param argument="--help" type="text" optional="false" label="help" help="show this help message and exit"/>
+</section>
+
+<section name="foo" title="foo" expanded="true">
```

Comment:
I like the conditional version much better, and it can be easily implemented

Reply:
We went the other way with `argparse2tool`, generating multiple xml files. It was a bit unclean in a way. Having a single large xml is probably easier to manage on the generation side, and maybe easier for users to work with. Let's see how the conditional works, I'm hopeful it's a lot nicer to work with too :)

Reply:
The conditional outputs are now functional, along with command generation. I haven't tested it yet in the galaxy because my env is kinda broken, but the output looks correct. I've also updated the tests so you can see how autopygen generates outputs in the updated version